### PR TITLE
MNIST download patch

### LINKED ISSUE #1737: Patch MNIST dataset downloading to fix CI
Currently, there is an issue with downloading MNIST dataset using torchvision. Let's introduce the following patch to our CI to fix it:
- https://github.com/pytorch/vision/issues/3500#issuecomment-790491487

Where to put that:
- create new step with the patch before [here](https://github.com/pytorch/ignite/blob/700f0e1325efc5dc0dce88d26284e51bc2a7c87c/.github/workflows/unit-tests.yml#L106)
- add patch [here](https://github.com/pytorch/ignite/blob/700f0e1325efc5dc0dce88d26284e51bc2a7c87c/.circleci/config.yml#L147).

### PR DESCRIPTION
**Pull request by trsvchn**
Fixes #1737

Description:

python base patch for MNIST download issue

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by trsvchn** [comment]
@vfdev-5 I am not sure about `circle ci` part and `wget` dependency for it

**Comment by vfdev-5** [comment]
> @vfdev-5 I am not sure about `circle ci` part and `wget` dependency for it

I think we have to install it for it and maybe GA as well

**Comment by trsvchn** [comment]
I'll check for GA

**Comment by trsvchn** [comment]
`wget` is fine for GA, at least for `ubuntu-latest`
Screenshot from 2021-03-04 14-51-30

**Comment by vfdev-5** [comment]
Yes, true! Just windows is failing to use it https://github.com/pytorch/ignite/pull/1738/checks?check_run_id=2031080498#step:12:24

**Comment by trsvchn** [comment]
Ok, well, our patch is not fully crossplatform, going to try smth different

**Comment by vfdev-5** [comment]
Maybe, we can try to install it with 
- https://github.com/webfolderio/wget-windows
- or using chocolatey action ?

**Comment by trsvchn** [comment]
Trying `chocolatey-action`

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -103,8 +103,8 @@ jobs:
           flags: cpu
           fail_ci_if_error: true
 
-      - name: Install wget on win
-        if: ${{ matrix.os }} == 'windows-latest'
+      - name: Install wget (Win)
```
Comment (vfdev-5) [inline .github/workflows/unit-tests.yml:106]:
```suggestion
      # Patch MNIST download: https://github.com/pytorch/ignite/issues/1737
      - name: Install wget (Win)
```

File: .github/workflows/unit-tests.yml [hunk 2]
```diff
@@ -103,6 +103,19 @@ jobs:
           flags: cpu
           fail_ci_if_error: true
 
+      - name: Install wget (Win)
+        if: matrix.os == 'windows-latest'
+        uses: crazy-max/ghaction-chocolatey@v1
+        with:
+          args: install wget
+
+      - name: Run MNIST Patch
```
Comment (vfdev-5) [inline .github/workflows/unit-tests.yml:112]:
```suggestion
      # Patch MNIST download: https://github.com/pytorch/ignite/issues/1737
      - name: Run MNIST Patch
```

**Review by vfdev-5 — APPROVED** [review]
Nit updates.
Thanks for the patch @trsvchn !

File: .circleci/config.yml [hunk 3]
```diff
@@ -146,6 +146,10 @@ jobs:
 
             # MNIST tests
 
+            # 0) mnist_patch.py
```
Comment (vfdev-5) [inline .circleci/config.yml:149]: @trsvchn I think we were too fast with circle ci. We have to install to there (linux machine)
```
FileNotFoundError: [Errno 2] No such file or directory: 'wget'
```
https://app.circleci.com/pipelines/github/pytorch/ignite/1483/workflows/8da2c054-e304-44de-9cf2-6d19ec0853d4/jobs/4243


Maybe, `apt-get update && apt-get install -y wget` can help there. Let me do that from my side.

Comment (trsvchn) [reply .circleci/config.yml:149]: I see, yes, let's try that!

Comment (vfdev-5) [reply .circleci/config.yml:149]: https://github.com/pytorch/ignite/pull/1739

**Pull request merged**